executor/builder.go

```diff
@@ -1631,7 +1631,8 @@ func (b *executorBuilder) buildApply(v *plannercore.PhysicalApply) Executor {
 		outerFilter, innerFilter = v.RightConditions, v.LeftConditions
 	}
 	tupleJoiner := newJoiner(b.ctx, v.JoinType, v.InnerChildIdx == 0,
-		defaultValues, otherConditions, retTypes(leftChild), retTypes(rightChild), nil)
+		defaultValues, otherConditions, retTypes(leftChild), retTypes(rightChild),
```

Comment:
`markChildrenUsedCols` is not used for expanding columns for the schema.
The schema decided by the optimizer is telling what the columns which the executor really needs. So the TableReader below the Join must follow the schema, but not return the extra column.

Reply:
That's true, `select a from table_30_utf8_4 order by a;` also breaks the code, resulted in index out of range, I'll look for a better fix.